bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/DMAssetPostProcessor.java

```diff
@@ -55,46 +55,53 @@
+                                        } else {
+                                            checkSetProperty(componentResource, Image.PN_IMAGE_SERVER_URL, null, list);
+                                        }
 
-                                Modification lastFileModification = getLastPropertyModification(request, list, "file");
-                                if ((lastFileModification != null) && (lastFileModification.getType() == ModificationType.DELETE)) {
-                                    checkSetProperty(request.getResource(), "smartCropRendition", null, list);
+                                        Modification lastFileModification = getLastPropertyModification(request, list, "file");
+                                        if ((lastFileModification != null) && (lastFileModification.getType() == ModificationType.DELETE)) {
+                                            checkSetProperty(componentResource, "smartCropRendition", null, list);
+                                        }
+                                    } else {
+                                        LOGGER.error("Unable to adapt resource '{}' used by image '{}' to an asset.", fileReference, componentResource.getPath());
+                                    }
+                                } else {
+                                    LOGGER.error("Unable to find resource '{}' used by image '{}'.", fileReference, componentResource.getPath());
                                 }
                             } else {
-                                LOGGER.error("Unable to adapt resource '{}' used by image '{}' to an asset.", fileReference, request.getResource().getPath());
+                                LOGGER.warn("File reference was null despite '{}' modification type.", lastFileReferenceModification.getType());
                             }
-                        } else {
-                            LOGGER.error("Unable to find resource '{}' used by image '{}'.", fileReference, request.getResource().getPath());
-                        }
-                    } else {
-                        LOGGER.warn("File reference was null despite '{}' modification type.", lastFileReferenceModification.getType());
+                            break;
+                        case DELETE:
+                            checkSetProperty(componentResource, Image.PN_IMAGE_SERVER_URL, null, list);
+                            checkSetProperty(componentResource, "smartCropRendition", null, list);
+                            break;
+                        default:
+                            //noop
                     }
-                    break;
-                case DELETE:
-                    checkSetProperty(request.getResource(), Image.PN_IMAGE_SERVER_URL, null, list);
-                    checkSetProperty(request.getResource(), "smartCropRendition", null, list);
-                    break;
-                default:
-                    //noop
+                }
```

Comment:
@oborysyu if component resource cannot be obtained, this is an unexpected condition, do LOGGER.warn

Reply:
done.